mlos_bench/mlos_bench/optimizers/convert_configspace.py

```diff
@@ -140,7 +143,10 @@ def _tunable_to_configspace(
     if tunable.quantization_bins:
         # Temporary workaround to dropped quantization support in ConfigSpace 1.0
         # See Also: https://github.com/automl/ConfigSpace/issues/390
-        monkey_patch_quantization(range_hp, tunable.quantization_bins)
+        new_meta = dict(range_hp.meta or {})
+        new_meta[QUANTIZATION_BINS_META_KEY] = tunable.quantization_bins
+        range_hp.meta = new_meta
+        monkey_patch_hp_quantization(range_hp)
```

Comment:
Part of me thinks this monkey patching could be entirely moved to the `mlos_core` and all we should have to do in `mlos_bench` is set `"quantization_bins"` in `meta` and let `mlos_core` handle the rest.

Reply:
@motus - the refinement I was thinking of

Reply:
Actually, I moved it just now into the `BaseOptimizer` class.  Have a look and lemme know what you think.  At a high level I think this works, but probably needs more tests.